notification/test/unit/google-function.spec.js

```diff
@@ -0,0 +1,118 @@
+const sinon = require('sinon')
+const chai = require('chai')
+const VError = require('verror')
+const { notificationTrigger } = require('../../index.googleFunction')
+const notificationHandler = require('../../src/handler/notification/notification.handler')
+const logger = require('../../src/utils/logger')
+
+const { expect } = chai
+const { getNotificationForTracking } = require('../../src/utils/commons')
+const {
+  buildMockErrorFromConcurrentModificaitonException,
+} = require('../test-utils')
+chai.use(require('chai-as-promised'))
+
+describe('Google Function handler', () => {
+  afterEach(() => {
+    notificationHandler.processNotification.restore()
+  })
+
+  const mockRequest = {
+    body: {
+      notificationItems: [
+        {
+          NotificationRequestItem: {
+            additionalData: {
+              'metadata.ctProjectKey': 'adyen-integration-test',
```

Comment:
Not sure your idea. The function processNotification itself requires project key as argument. 
https://github.com/commercetools/commercetools-adyen-integration/blob/81b503d3bad5eac2bc74b5bf1f407e235a0756e2/notification/src/handler/notification/notification.handler.js#L14

The test case works as long as the given project key is configured in environment, otherwise an unrecoverable error is throw, which fails the test csae for sure.

Reply:
> The test case works as long as the given project key is configured in environment, otherwise an unrecoverable error is throw..

If it's the case my assumption is right, In general unit tests should not require any additional setup to run it, it means in local if you don't have the same env variable configured, the test will fail, which is not good. To me it looks like a dependency in unit test to env variable.

https://github.com/commercetools/commercetools-adyen-integration/blob/939fcd9b112555a68f82237f57924fcea9dc808e/extension/package.json#L12

For instance in extension module's package json the unit tests are configured to mock-default-config.js which sets a mock env variable. this does not seem a case for notification module.

Reply:
Use mock CTP config and Adyen config in unit test so that dummy project key and dummy adyen merchant code can be used.
https://github.com/commercetools/commercetools-adyen-integration/pull/747/commits/bc2c1ad94c86a174c36b4a25696ded0610bd6063